promql/functions.go

```diff
@@ -1218,6 +1238,78 @@ func (s *vectorByReverseValueHeap) Pop() interface{} {
 	return el
 }
 
+type vectorByLabelHeap struct {
+	vector Vector
+	labels []string
+}
+
+func (s vectorByLabelHeap) Len() int {
+	return len(s.vector)
+}
+
+func (s vectorByLabelHeap) Less(i, j int) bool {
+	// Iterate over each given label
+	for k := 0; k < len(s.labels); k++ {
```

Comment:
This would be more Go-y:

```suggestion
	for _, label := range s.labels {
```